import_export/admin.py

```diff
@@ -217,10 +217,6 @@ def read_from_tmp_storage(self, tmp_storage, input_format):
         else:
             data = tmp_storage.read(input_format.get_read_mode(), encoding=self.from_encoding)
 
-            # if the object is being read from a cache, then it might be binary
-            # if so, decode before returning
-            if isinstance(data, bytes):
-                data = force_str(data, encoding=self.from_encoding)
```

Comment:
Hi @matthewhegarty, I tried moving the decode to the tmp_storage classes. See below.